File: spring-aws-s3-java/src/main/java/com/github/migangqui/spring/aws/s3/service/AmazonS3Service.java
package com.github.migangqui.spring.aws.s3.service;

import com.github.migangqui.spring.aws.s3.bean.*;

import java.io.InputStream;
import java.util.concurrent.Future;

public interface AmazonS3Service {
	
	UploadFileResponse uploadFile(UploadFileRequest request);

	Future<UploadFileResponse> uploadFileAsync(UploadFileRequest request);

	GetFileResponse getFile(GetFileRequest request);
	
	DeleteFileResponse deleteFile(DeleteFileRequest request);

}

File: spring-aws-s3-kotlin/src/main/kotlin/com/github/migangqui/spring/aws/s3/service/AmazonS3Service.kt
package com.github.migangqui.spring.aws.s3.service

import com.github.migangqui.spring.aws.s3.bean.*
import java.io.InputStream
import java.util.concurrent.Future

interface AmazonS3Service {
    fun uploadFile(request: UploadFileRequest): UploadFileResponse

    fun uploadFileAsync(request: UploadFileRequest): Future<UploadFileResponse>

    fun getFile(request: GetFileRequest): GetFileResponse

    fun deleteFile(request: DeleteFileRequest): DeleteFileResponse
}